demux/packet.c

```diff
@@ -51,9 +52,18 @@ static void packet_destroy(void *ptr)
     demux_packet_unref_contents(dp);
 }
 
-static struct demux_packet *packet_create(void)
+static struct demux_packet *packet_create(struct demux_packet_pool *pool)
 {
-    struct demux_packet *dp = talloc(NULL, struct demux_packet);
+    struct demux_packet *dp = pool ? demux_packet_pool_pop(pool) : NULL;
+    struct AVPacket *avpkt = NULL;
+    if (dp) {
+        avpkt = dp->avpacket;
+        if (avpkt)
+            av_packet_unref(avpkt);
+        ta_free_children(dp);
```

Comment:
so as far as I can see we unref and free what I assume to be buffers that were referenced from the `avpkt`.
since this happens unconditionally shouldn't this happen upon pushing to the pool, to free the memory?

Reply:
This is intended. `packet_create` is called in relatively sane intervals and rate for demuxing a video, we can get the single packet from pool and initialize/clear it before use. What this avoids is the issue where we have huge demuxer cache filled and suddenly we want to "clear all now", this happens on file close for example. In this case we don't want to iterate over all packets and distribute this load over the time while the packets are reused for next playback.

Reply:
Okay, then I suggest documenting this with comments and moving the freeing to `demux_packet_pool_pop` so that it is documented to return an empty packet with (optionally) an empty avpacket on the side.

Reply:
Done.